### ISSUE TITLE
Saving and quitting in TCL caused destroyed floor to grow a roof

### ISSUE THREAD

**Issue by Terrorforge** [CleverRaven/Cataclysm-DDA#54457]
### Describe the bug

I was in a TCL, specifically the "lab security area", where a couple of hulks on the floor below had torn down several tiles of the floor on my level. I wanted to ensure I had a proper save before I did something stupid, so I did a "save and quit" form the menu and then loaded back in, only to find that most of the hole had been plugged up by "flat roof"

### Steps to reproduce

1. Let some hulks wreck the roof
2. Save and quit
3. Load

### Expected behavior

Buildings should not grow new parts. At least not unintentionally.

### Screenshots

![image](https://user-images.githubusercontent.com/5107449/149638160-500f7740-de7c-4c82-986b-043b78c4be56.png)

### Versions and configuration

- OS: Windows
    - OS Version: 10.0 2009
- Game Version: 7ddf235 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions],
    Bionic Slots [cbm_slots],
    Extra Mutated Scenarios [extra_mut_scens]
]

### Additional context

I'm trying to provide a zipped save file, but when I try I just get an error that says "is not included in list".

**Comment by LeahLuong** [comment]
When the website says, "is not included in the list," it might not actually mean that. It may be you aren't providing a recognized format but it might also mean your file is too large. Upwards of 50 MB seems too large.

**Comment by hexagonrecursion** [comment]
This is not specific to TCL

![DeepinScreenshot_select-area_20220116110155](https://user-images.githubusercontent.com/52621858/149652123-9d745b7a-e16d-456a-a432-a3b6aa89c279.png)
![DeepinScreenshot_select-area_20220116110236](https://user-images.githubusercontent.com/52621858/149652124-b7b0b9cc-db77-4f8b-a544-05f8c90792bb.png)

**Comment by Terrorforge** [comment]
> When the website says, "is not included in the list," it might not actually mean that. It may be you aren't providing a recognized format but it might also mean your file is too large. Upwards of 50 MB seems too large.

That would do it. The save is well over a gigabyte, so even the zipped file is something like 150mb.

Thankfully hexagonrecursion has confirmed the bug, so the save shouldn't be necessary.

**Comment by LeahLuong** [comment]
This may be related to #53549. Same process where reloading fills in terrain gaps. Possible dupe.

**Comment by hexagonrecursion** [comment]
> Possible dupe.

Nope. In #53549 the issue is that there are gaps where there should not be. This is *almost* the opposite: gaps disappear.
